### PR TITLE
Gossip lightclient

### PR DESCRIPTION
**Pull request by enriavil1**
Added handlers (they only log) and subscribed to the public gossip

### CODE REVIEW

File: cmd/lightclient/sentinel/sentinel.go [hunk 1]
```diff
@@ -25,18 +25,23 @@ import (
...
 	"github.com/ledgerwatch/log/v3"
 	"github.com/libp2p/go-libp2p"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
 	"github.com/libp2p/go-libp2p/core/host"
+	"github.com/libp2p/go-libp2p/core/peer"
 	"github.com/libp2p/go-libp2p/p2p/protocol/identify"
 	"github.com/pkg/errors"
 )
 
+var disconnectPeerCh = make(chan peer.ID, 0)
```
Comment (Giulio2002) [inline cmd/lightclient/sentinel/sentinel.go:35]: could cause a deadlock also bad global variable.

File: cmd/lightclient/sentinel/sentinel.go [hunk 2]
```diff
@@ -114,9 +119,36 @@ func (s *Sentinel) createListener() (*discover.UDPv5, error) {
 		return nil, err
 	}
 
+	go func() {
```
Comment (Giulio2002) [inline cmd/lightclient/sentinel/sentinel.go:122]: We can merge this but I do not like this archittecture, I may change it in the future.

File: cmd/lightclient/sentinel/handlers.go [hunk 3]
```diff
@@ -47,6 +70,56 @@ func pingHandler(stream network.Stream) {
...
+		log.Warn("Goodbye handler crashed", "err", err)
+		return
+	}
+
+	if n != 8 {
+		log.Warn("Could not send Goodbye")
+	}
+
+	peerId := stream.Conn().RemotePeer()
+	disconnectPeerCh <- peerId
```
Comment (Giulio2002) [inline cmd/lightclient/sentinel/handlers.go:105]: again I do not like this, this is asking for troubles, possibility of deadlocks.

**Comment by Giulio2002** [comment]
I will merge it and fix up the peer disconnection logic to something safer.

**Comment by AskAlexSharov** [comment]
fixed linter errors by: https://github.com/ledgerwatch/erigon/pull/5561/files#diff-f422a5f3ab817cec1e2a059e4714fdbbce8be457cccc8c888461be8e3a27f8a8

**Pull request merged**